### ISSUE TITLE
Name Consistency & Refactoring Ideas

### ISSUE THREAD

**Issue by LordFokas** [NanoHttpd/nanohttpd#242]
As I mentioned earlier in this PR: #241

... Server class names are inconsistent.
The HTTP server is called NanoHTTPD, but the WebSocket server is called NanoWebSocketServer.
It seems to me it would make a lot more sense to call it NanoWSD, for several reasons:

1) The names become consistent.
2) Protocol name is right there on the server names.
3) Name becomes shorter. Doesn't really matter, we all see class names that look like trains every now and then, but shorter is generaly nicer :)

**HOWEVER...**
This is a community-driven project, so I'm not going to create a pull request for such a superficial thing (at least for some, but definitely not for me, I totally obsess over things like package, class and method names) without some consensus. Plus, and the real reason, I don't know how far that name goes between all the subclasses, test suites, and all that bureaucracy with XML files from build and integration systems, and I don't really feel like breaking anything... yet.

**TL;DR:** Can we change the name of the WebSocket server from NanoWebSocketServer to NanoWSD, please? :)

**Comment by LordFokas** [comment]
@ritchieGitHub , @elonen 
Based on your replies on my previous PR, which is referenced above, I'm glad you've approved my idea for the name change, and I'd like to take the chance to discuss some more refactoring before doing anything with it, namely:

- Move /websocket/src/main/java/ **fi/iki/elonen/NanoWebSocketServer.java** (NanoWSD) into core/src/main/java/... along with NanoHTTPD.
- Move all unit tests, resources, samples and everything else related to NanoWSD into the same places as NanoHTTPD's, with separate sub-packages for HTTP and WS.
- Make a lot more stuff protected instead of private in both NanoHTTPD and NanoWSD. Details on it later, but the core idea is, as referenced in #236, making the project more easily extendable.
- Change the way NanoWSD handles WebSockets internally: either use Generics to define the WebSocket class our implementation of NanoWSD uses **or** create protected handling methods in the WebSocket class that by default redirect calls to NanoWSD. This is not strictly necessary but it's ridiculously useful for me to be able to have each WebSocket define how it handles events, and more people might think the same.

I'd also like to pull @psh into this discussion, if he's not too busy.

Now, what do you think of what I just proposed?
I'm OK with handling most or all of it, but I'm a bit afraid to break anything. I'm not exactly versed in things like Unit Tests / Maven / Gradle / etc, so I don't know how far some of those changes go, but provided you don't mind to guide me a bit I can handle this :)

And thanks for wasting your time reading my long issues.

**Comment by ritchieGitHub** [comment]
I thinkl merging the modules is wrong, NanoWSD is just one usecase of nanohttpd and should be kept separate.  @psh is not involved anymore.

Refactoring the NanoWSD is a good thing, specially because it still has minimal unit test coverage.

But more important would be reaching a bigger test coverage, with unit tests.
